File: Queue/Q1_NumberofStudentsUnabletoEatLunch/Q1_NumberofStudentsUnabletoEatLunch.java
class Solution {                     // Define the Solution class
    public int countStudents(int[] students, int[] sandwiches) { // Method to count students who can't eat
        
        int[] counts = new int[2];    // Array to count students who like type 0 and type 1
        
        for (int student : students)  // Loop through each student
            counts[student]++;        // Increase count of the student's preference
        
        int remaining = sandwiches.length; // Total number of sandwiches
        
        for (int sandwich : sandwiches) {  // Loop through each sandwich
            if (counts[sandwich] == 0) break; // Stop if no student wants this sandwich
            if (remaining-- == 0) break;     // Reduce remaining count and stop if none left
            counts[sandwich]--;              // One student takes the sandwich
        }
        
        return remaining;             // Return number of students who couldn't eat
    }
}

File: Queue/Q1_NumberofStudentsUnabletoEatLunch/Q1_NumberofStudentsUnabletoEatLunch.py
class Solution(object):
    def countStudents(self, students, sandwiches):
        st0 = students.count(0)  # Count number of 0s
        st1 = len(students) - st0  # Remaining students are 1s

        for x in sandwiches:
            if x == 1:  # square sandwich
                if st1 > 0:         #If there are students who want a square sandwich
                    st1 -= 1        #allocate to him and remove him

                else:               #If no student wants a square sandwich, No other sandwich 
                    return st0+st1  #in the stack can be allocated without allocating this sandwich
                                    
                    
            else:       #circular sandwich
                if st0 > 0:         # If there are students who wants circular sandwich
                    st0 -= 1        # allocate to him and remove him

                else:               # If no student wants a circular sandwich, No other sandwich 
                    return st0+st1  # in the stack can be allocated without allocating this sandwich

        return 0  # All sandwiches distributed successfully